src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/VirtualMethodIndexData.cs

```diff
@@ -6,9 +6,37 @@
 namespace Microsoft.Interop
 {
     /// <summary>
-    /// VirtualMethodIndexAttribute data
+    /// Contains the data related to a VirtualMethodIndexAttribute, without references to Roslyn symbols.
+    /// See <seealso cref="VirtualMethodIndexCompilationData"/> for a type with a reference to the StringMarshallingCustomType
     /// </summary>
-    internal sealed record VirtualMethodIndexData(int Index) : InteropAttributeCompilationData
+    internal sealed record VirtualMethodIndexData(int Index) : InteropAttributeData
+    {
+        public bool ImplicitThisParameter { get; init; }
+
+        public MarshalDirection Direction { get; init; }
+
+        public bool ExceptionMarshallingDefined { get; init; }
+
+        public ExceptionMarshalling ExceptionMarshalling { get; init; }
```

Comment:
Can't we just define all these in the ctor for the type? We are retyping things a lot below when it seems like we need all of them always.